src/jdk.internal.vm.ci/share/classes/jdk.vm.ci.hotspot.aarch64/src/jdk/vm/ci/hotspot/aarch64/AArch64HotSpotRegisterConfig.java

```diff
@@ -265,11 +268,25 @@ private CallingConvention callingConvention(RegisterArray generalParameterRegist
 
             if (locations[i] == null) {
                 ValueKind<?> valueKind = valueKindFactory.getValueKind(kind);
+                int kindSize = valueKind.getPlatformKind().getSizeInBytes();
+                if (macOS && currentStackOffset % kindSize != 0) {
+                    // In MacOS natural alignment is used
+                    // See https://developer.apple.com/documentation/xcode/writing-arm64-code-for-apple-platforms
+                    currentStackOffset += kindSize - currentStackOffset % kindSize;
+                }
                 locations[i] = StackSlot.get(valueKind, currentStackOffset, !type.out);
-                currentStackOffset += Math.max(valueKind.getPlatformKind().getSizeInBytes(), target.wordSize);
+                if (macOS) {
+                    // In MacOS "Function arguments may consume slots on the stack that are not multiples of 8 bytes"
+                    // See https://developer.apple.com/documentation/xcode/writing-arm64-code-for-apple-platforms
+                    currentStackOffset += kindSize;
+                } else {
+                    currentStackOffset += Math.max(kindSize, target.wordSize);
+                }
             }
         }
-
+        if (currentStackOffset % target.stackAlignment != 0) {
```

Comment:
I think this must be taken care of somewhere else. Otherwise, amd64 would also have this same problem, as its stack alignment is also 16 bytes

Reply:
The SIGBUS issue looks to be more a problem with AArch64TestAssembler.java rather than an issue here. It would be best to align stack size at [        emitGrowStack(cc.getStackSize());](https://github.com/openjdk/jdk/blob/44fe958c8a924dda436125716b164a335199e22b/test/hotspot/jtreg/compiler/jvmci/jdk.vm.ci.code.test/src/jdk/vm/ci/code/test/aarch64/AArch64TestAssembler.java#L272) and         [emitGrowStack(-cc.getStackSize()); ](https://github.com/openjdk/jdk/blob/44fe958c8a924dda436125716b164a335199e22b/test/hotspot/jtreg/compiler/jvmci/jdk.vm.ci.code.test/src/jdk/vm/ci/code/test/aarch64/AArch64TestAssembler.java#L282)

The same thing could be done to AMD64TestAssembler. 

At least for graal, this more closely matches the logic of [AArchFrameMap](https://github.com/oracle/graal/blob/bf17700b2593f7e4a2c71f465d53a96a38f92048/compiler/src/org.graalvm.compiler.lir.aarch64/src/org/graalvm/compiler/lir/aarch64/AArch64FrameMap.java#L100) and [AMD64FrameMap](https://github.com/oracle/graal/blob/bf17700b2593f7e4a2c71f465d53a96a38f92048/compiler/src/org.graalvm.compiler.lir.amd64/src/org/graalvm/compiler/lir/amd64/AMD64FrameMap.java#L109)